core/vm/actor/builtin/v2/payment_channel/payment_channel_actor.cpp

```diff
@@ -0,0 +1,161 @@
+    if (code != kAccountCodeCid) {
+      return VMExitCode::kErrForbidden;
+    }
+    return std::move(resolved);
+  }
+
+  outcome::result<State> assertCallerInChannel(Runtime &runtime) {
+    OUTCOME_TRY(state, runtime.getCurrentActorStateCbor<State>());
+    if (runtime.getImmediateCaller() != state.from
+        && runtime.getImmediateCaller() != state.to) {
+      return VMExitCode::kErrForbidden;
+    }
+    return std::move(state);
+  }
+
+  ACTOR_METHOD_IMPL(Construct) {
+    if (runtime.getImmediateCaller() != kInitAddress) {
+      return VMExitCode::kSysErrForbidden;
+    }
+    OUTCOME_TRY(to, resolveAccount(runtime, params.to));
+    OUTCOME_TRY(from, resolveAccount(runtime, params.from));
+
+    State state{from, to, 0, 0, 0, {}};
+    IpldPtr {
+      runtime
+    }
+    ->load(state);
+    OUTCOME_TRY(runtime.commitState(state));
+    return fc::outcome::success();
+  }
+
+  ACTOR_METHOD_IMPL(UpdateChannelState) {
+    OUTCOME_TRY(readonly_state, assertCallerInChannel(runtime));
+    const auto &voucher = params.signed_voucher;
+    if (!voucher.signature) {
+      return VMExitCode::kErrIllegalArgument;
+    }
+    if (readonly_state.settling_at != 0
+        && runtime.getCurrentEpoch() >= readonly_state.settling_at) {
+      return VMExitCode::kErrFirstActorSpecificExitCode;
+    }
+    if (params.secret.size() > kMaxSecretSize) {
+      return VMExitCode::kErrIllegalArgument;
+    }
+    auto voucher_signable = voucher;
+    voucher_signable.signature = boost::none;
+    OUTCOME_TRY(voucher_signable_bytes, codec::cbor::encode(voucher_signable));
+    auto &signer = runtime.getImmediateCaller() != readonly_state.to
+                       ? readonly_state.to
+                       : readonly_state.from;
+    OUTCOME_TRY(verified,
+                runtime.verifySignature(
+                    *voucher.signature, signer, voucher_signable_bytes));
+    if (!verified) {
+      return VMExitCode::kErrIllegalArgument;
+    }
+    const auto paych_addr = runtime.getCurrentReceiver();
+    OUTCOME_TRY(voucher_addr, runtime.resolveAddress(voucher.channel));
+    if (paych_addr != voucher_addr) {
+      return VMExitCode::kErrIllegalArgument;
+    }
+    if (runtime.getCurrentEpoch() < voucher.time_lock_min) {
+      return VMExitCode::kErrIllegalArgument;
+    }
+    if (voucher.time_lock_max != 0
+        && runtime.getCurrentEpoch() > voucher.time_lock_max) {
+      return VMExitCode::kErrIllegalArgument;
+    }
+    if (voucher.amount.sign() < 0) {
+      return VMExitCode::kErrIllegalArgument;
+    }
+    if (!voucher.secret_preimage.empty()
+        && gsl::make_span(runtime.hashBlake2b(params.secret))
+               != gsl::make_span(voucher.secret_preimage)) {
+      return VMExitCode::kErrIllegalArgument;
+    }
+    if (voucher.extra) {
+      OUTCOME_TRY(params_extra, encodeActorParams(voucher.extra->params));
+      OUTCOME_TRY(runtime.send(
+          voucher.extra->actor, voucher.extra->method, params_extra, 0));
```

Comment:
```suggestion
      OUTCOME_TRY(runtime.send(
          voucher.extra->actor, voucher.extra->method, voucher.extra->params, 0));
```
https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/paych/paych_actor.go#L216

Reply:
fixed